borgmatic/hooks/postgresql.py

```diff
@@ -56,9 +56,11 @@ def dump_databases(databases, log_prefix, location_config, dry_run):
         )
         all_databases = bool(name == 'all')
         dump_format = database.get('format', 'custom')
+        default_dump_command = 'pg_dumpall' if all_databases else 'pg_dump'
+        dump_command = database.get('pg_dump_command') or default_dump_command
         command = (
             (
-                'pg_dumpall' if all_databases else 'pg_dump',
+                dump_command,
```

Comment:
The only problem I see with this is that the user won't be able to specify separate `pg_dump` and `pg_dumpall` commands. So you could in theory configure an "all" database hook with no way to dump it. What do you think of having a separate `pg_dumpall_command` configuration option to handle this case?

Reply:
hey there! it started like that, with `pg_dump_command` and `pg_dumpall_command`, but then I realised they're never needed for the same `postgresql_databases` item. Using one or the other depends on having `name: my_database` or `name: all`.

With that in mind I went with different semantics: "I don't want borgmatic to choose depending on the value under `name:` and instead I want to provide my own command".

Other possibility that I considered was a _total override_: "let the user provide the exact command line plus switches, with placeholders for name, user, host, port, ..."
It's more verbose which is why I didn't go with that one, but that's just because I'm lazy.

Am I missing something? What do you think?